golang/Dockerfile

```diff
@@ -9,7 +34,15 @@ ENV GIT_SHA256 20acda973eca1df056ad08bec6e05c3136f40a1b90e2a290260dfc36e9c2c800
 RUN Invoke-WebRequest -UseBasicParsing $env:GIT_DOWNLOAD_URL -OutFile git.zip; \
     if ((Get-FileHash git.zip -Algorithm sha256).Hash -ne $env:GIT_SHA256) {exit 1} ; \
     Expand-Archive git.zip -DestinationPath C:\git; \
-    Remove-Item git.zip; \
-    Write-Host 'Updating PATH ...'; \
-    $env:PATH = 'C:\git\cmd;C:\git\mingw64\bin;C:\git\usr\bin;' + $env:PATH; \
-    Set-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment\' -Name Path -Value $env:PATH
+    Remove-Item git.zip
+
+FROM microsoft/nanoserver-insider
+
+COPY --from=builder /go /go
+COPY --from=builder /git /git
+
+# ideally, this would be C:\go to match Linux a bit closer, but C:\go is the recommended install path for Go itself on Windows
+ENV GOPATH C:\\gopath
+WORKDIR $GOPATH
+
+RUN setx PATH "%PATH%;C:\%GOPATH%\bin;C:\go\bin;C:\git\cmd;C:\git\mingw64\bin;C:\git\usr\bin"
```

Comment:
setx /M PATH does not work as nanoserver-insider does not run as ContainerAdministrator.